lib/_http_server.js

```diff
@@ -1022,7 +1022,18 @@ function parserOnIncoming(server, socket, state, req, keepAlive) {
 
   let handled = false;
 
+
   if (req.httpVersionMajor === 1 && req.httpVersionMinor === 1) {
+
+    // From RFC 7230 5.4 https://datatracker.ietf.org/doc/html/rfc7230#section-5.4
+    // A server MUST respond with a 400 (Bad Request) status code to any
+    // HTTP/1.1 request message that lacks a Host header field
+    if (req.headers.host === undefined) {
```

Comment:
This could end up being a fairly significant breaking change. Perhaps we should make this behavior configurable? On by default but able to switch it off via an option.

Reply:
Good point. I agree.